tests/test_inventory_builder.py

```diff
@@ -74,3 +99,174 @@ def test_expr_reso_grammar_ok(inv_builder, test_data, arg, exp):
 
     t = Transformer(data=test_data)
     assert t.transform(p.parse(arg)) == exp
+
+
+@pytest.mark.parametrize("args", [
+    ({  # config file plus list
+        "config-file": "netbox.yml",
+        "list": True,
+        "host": None,
+    }),
+    ({  # no list, plus host
+        "config-file": "netbox.yml",
+        "list": False,
+        "host": "hostname",
+    }),
+    ({  # list alone, no host, no config file
+        "config-file": "netbox.yml",
+        "list": True,
+        "host": None,
+    }),
+    ({  # list alone, no host, no config file
+        "config-file": "netbox.yml",
+        "list": False,
+        "host": None,
+    }),
+])
+def test_init_builder_ok(args, config):
+    class Args:
+        def __init__(self):
+            self.config_file = args['config-file']
+            self.host = args['host']
+            self.list = args['list']
+
+    builder = InventoryBuilder(Args(), config)
+
+    assert builder.config_file == args['config-file']
+    assert builder.list_mode == args['list']
+    assert builder.host == args['host']
+
+
+@pytest.mark.parametrize("hostvalue", [
```

Comment:
I would suggest another structure for parametrized arguments:
two arguments :  "arg,expected", where arg would contain a dict with a null name (or  not), and controlling the function returns what is in 'expected'